udata/core/topic/apiv2.py

```diff
@@ -0,0 +1,269 @@
+    @apiv2.expect([topic_add_items_fields])
+    @apiv2.marshal_with(topic_fields)
+    @apiv2.response(400, 'Malformed object id(s) in request')
+    @apiv2.response(400, 'Expecting a list')
+    @apiv2.response(400, 'Expecting a list of dicts with id attribute')
+    @apiv2.response(404, 'Topic not found')
+    @apiv2.response(404, 'Dataset(s) not found')
+    @apiv2.response(403, 'Forbidden')
+    def post(self, topic):
+        '''Add datasets to a given topic from a list of dataset ids'''
+        def add_dataset(topic, dataset):
+            if dataset not in topic.datasets:
+                topic.datasets.append(dataset)
+            return topic
+
+        def get_dataset(dataset_id):
+            try:
+                dataset = Dataset.objects.get_or_404(id=dataset_id)
+            except mongoengine.errors.ValidationError:
+                apiv2.abort(400, 'Malformed object id(s) in request')
+            return dataset
+
+        if not TopicEditPermission(topic).can():
+            apiv2.abort(403, 'Forbidden')
+
+        data = request.json
+
+        if not isinstance(data, list):
+            apiv2.abort(400, 'Expecting a list')
+        if not all(isinstance(d, dict) and d.get('id') for d in data):
+            apiv2.abort(400, 'Expecting a list of dicts with id attribute')
+
+        datasets = (get_dataset(d['id']) for d in data)
+        for dataset in datasets:
+            topic = add_dataset(topic, dataset)
+        topic.save()
+
+        # TODO: maybe we should return None, or the topics/datasets page
+        # but pagination might not match
+        return topic, 201
+
+
+@ns.route('/<topic:topic>/datasets/<dataset:dataset>/', endpoint='topic_dataset', doc={
+    'params': {'topic': 'The topic ID', 'dataset': 'The dataset ID'}
+})
+class TopicDatasetAPI(API):
+    @apiv2.secure
+    @apiv2.response(404, 'Topic not found')
+    @apiv2.response(404, 'Dataset not found in topic')
+    @apiv2.response(204, 'Success')
+    def delete(self, topic, dataset):
+        '''Delete a given dataset from the given topic'''
+        if not TopicEditPermission(topic).can():
+            apiv2.abort(403, 'Forbidden')
+
+        if dataset.id not in (d.id for d in topic.datasets):
+            apiv2.abort(404, 'Dataset not found in topic')
+        topic.datasets = [d for d in topic.datasets if d.id != dataset.id]
+        topic.save()
+
+        return None, 204
+
+
+@ns.route('/<topic:topic>/reuses/', endpoint='topic_reuses', doc=common_doc)
+class TopicReusesAPI(API):
+    @apiv2.doc('topic_reuses')
+    @apiv2.expect(reuse_parser.parser)
+    @apiv2.marshal_with(reuse_page_fields)
+    def get(self, topic):
+        '''Get a given topic reuses, with filters'''
+        args = reuse_parser.parse()
+        reuses = Reuse.objects(deleted=None, private__ne=True).filter(
+            id__in=[d.id for d in topic.reuses]
+        )
+        # warning: topic in reuse_parser is different from Topic
+        args.pop('topic', None)
```

Comment:
@maudetes :-/

Reply:
Actually I don't need to pop it, it will filter properly on topicoThematique of the reuse. But how ugly this is 🙈

Reply:
Oh my, quite ugly indeed! We'll need to fix this semantic issue shortly